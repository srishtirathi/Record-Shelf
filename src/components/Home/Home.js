import React from 'react';
import './Home.css';
import { Link } from 'react-router-dom';

const Home = () => (
  <div className="home">
    <h4>Seems a bit empty in here :(</h4>
    <Link to="/allrecords"><button type="button">Sync</button></Link>
  </div>
);
export default Home;
