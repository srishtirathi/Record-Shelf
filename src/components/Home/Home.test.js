/* eslint-disable react/jsx-props-no-spreading */
import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import {
  BrowserRouter,
} from 'react-router-dom';
import Home from './Home';

describe(Home.name, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('sanity check', () => {
    expect(1 + 1).toBe(2);
  });
  it('should match snapshot', () => {
    const { container } = render(<BrowserRouter><Home /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });
  it('should render text and button', () => {
    render(<BrowserRouter><Home /></BrowserRouter>);
    screen.getByText('Seems a bit empty in here :(');

    screen.getByText('Sync');
  });
  test('Should take to AllRecords page on click', () => {
    render(<BrowserRouter><Home /></BrowserRouter>);
    const homeLink = screen.getByText('Sync');
    fireEvent.click(homeLink);
    expect(document.location.href).toBe('http://localhost/allrecords');
  });
});
