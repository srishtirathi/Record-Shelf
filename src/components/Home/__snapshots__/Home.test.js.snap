// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div>
  <div
    class="home"
  >
    <h4>
      Seems a bit empty in here :(
    </h4>
    <a
      href="/allrecords"
    >
      <button
        type="button"
      >
        Sync
      </button>
    </a>
  </div>
</div>
`;
